subsys/net/lib/coap/coap.c

```diff
@@ -70,60 +70,82 @@ static int get_coap_packet_len(struct net_pkt *pkt)
 	return len;
 }
 
-static int decode_delta(u16_t num, struct option_context *context,
-			u16_t *decoded)
+static int check_frag_read_status(const struct net_buf *frag, u16_t offset)
 {
-	int hdrlen = 0;
+	if (!frag && offset == 0xffff) {
+		return -EINVAL;
+	} else if (!frag && offset == 0) {
+		return 0;
+	}
+
+	return 1;
+}
 
-	if (num == 13) {
+static int decode_delta(struct option_context *context, u16_t opt,
+			u16_t *opt_ext, u16_t *hdr_len)
+{
+	int ret = 0;
+
+	if (opt == 13) {
```

Comment:
There are lot of magic values in this PR. This is very confusing to read actually, these values need to be explained somewhere. Please put these values in defines and document what they mean.

Reply:
Ok, I will add some information.